<launch>
  <!-- size of map, change the size in x, y, z according to your application -->
  <arg name="map_size_x" value="50.0"/>
  <arg name="map_size_y" value="50.0"/>
  <arg name="map_size_z" value="10.0"/>

  <!-- Global odom prefix -->
  <arg name="odom_prefix" value="/state_ukf/odom" />
  <arg name="drone_num" value="1" />

  <!-- Map generation -->
  <!-- transformed_hills.pcd / pillar.pcd / transformed_house1.pcd / office.pcd / transformed_cloud.pcd -->
  <node pkg ="map_generator" name ="map_pub" type ="map_pub" output = "screen" args="$(find map_generator)/resource/transformed_village.pcd"> 
  <!-- <node pkg ="map_generator" name ="map_pub" type ="map_pub" output = "screen" args="$(find map_generator)/resource/transformed_village01.pcd">  -->
  </node>

  <!-- uav group-->
  <!-- uav_mode: 1-high uav  0-low uav -->
  <include file="$(find exploration_manager)/launch/single_drone_exploration.xml">
    <arg name="uav_mode" value="1"/>
    <arg name="drone_id" value="1"/>
    <arg name="init_x"   value="0"/>
    <arg name="init_y"   value="0"/>
    <arg name="init_z"   value="3.9"/> 
    <arg name="map_size_x" value="$(arg map_size_x)"/>
    <arg name="map_size_y" value="$(arg map_size_y)"/>
    <arg name="map_size_z" value="$(arg map_size_z)"/>      
    <arg name="odom_prefix" value="$(arg odom_prefix)"/>
    <arg name="drone_num" value="$(arg drone_num)"/>
    <arg name="simulation" default="true"/>
  </include>

  <!-- <include file="$(find exploration_manager)/launch/single_drone_exploration.xml">
    <arg name="uav_mode" value="1"/>
    <arg name="drone_id" value="2"/>
    <arg name="init_x"   value="0"/>
    <arg name="init_y"   value="-1"/>
    <arg name="init_z"   value="3.9"/> 
    <arg name="map_size_x" value="$(arg map_size_x)"/>
    <arg name="map_size_y" value="$(arg map_size_y)"/>
    <arg name="map_size_z" value="$(arg map_size_z)"/>
    <arg name="odom_prefix" value="$(arg odom_prefix)"/>
    <arg name="drone_num" value="$(arg drone_num)"/>
    <arg name="simulation" default="true"/>
  </include>

    <include file="$(find exploration_manager)/launch/single_drone_exploration.xml">
    <arg name="uav_mode" value="1"/>
    <arg name="drone_id" value="3"/>
    <arg name="init_x"   value="-1"/>
    <arg name="init_y"   value="-1"/>
    <arg name="init_z"   value="3.9"/> 
    <arg name="map_size_x" value="$(arg map_size_x)"/>
    <arg name="map_size_y" value="$(arg map_size_y)"/>
    <arg name="map_size_z" value="$(arg map_size_z)"/>
    <arg name="odom_prefix" value="$(arg odom_prefix)"/>
    <arg name="drone_num" value="$(arg drone_num)"/>
    <arg name="simulation" default="true"/>
  </include> -->

  <!-- <include file="$(find exploration_manager)/launch/single_drone_exploration.xml">
    <arg name="uav_mode" value="0"/>
    <arg name="drone_id" value="3"/>
    <arg name="init_x"   value="0"/>
    <arg name="init_y"   value="0"/>
    <arg name="init_z"   value="1"/>
    <arg name="map_size_x" value="$(arg map_size_x)"/>
    <arg name="map_size_y" value="$(arg map_size_y)"/>
    <arg name="map_size_z" value="$(arg map_size_z)"/>
    <arg name="odom_prefix" value="$(arg odom_prefix)"/>
    <arg name="drone_num" value="$(arg drone_num)"/>
    <arg name="simulation" default="true"/>
  </include> -->

</launch>
